cpp/src/arrow/memory_pool.h

```diff
@@ -144,6 +152,16 @@ class ARROW_EXPORT ProxyMemoryPool : public MemoryPool {
 /// Return the process-wide default memory pool.
 ARROW_EXPORT MemoryPool* default_memory_pool();
 
+ARROW_EXPORT MemoryPool* system_memory_pool();
+
+#ifdef ARROW_JEMALLOC
```

Comment:
Note this is changing the library API depending on the compilation options. I don't know if that's desirable (though it would only be a problem if calling one of those functions).

Reply:
@xhochy What do you think?

Reply:
I would rather return `null` or status instead of changing ABI based on compilation options. This makes it hard for binaries to built against Arrow and also use the different allocators explicitly.

Reply:
I like @xhochy 's idea.

Reply:
Ok, I will make this unconditional and return a Status.